### PR TITLE
Solc

### PR DESCRIPTION
**Pull request by iostat**
summer's SOLCstice is almost here!

this adds a vague notion of a "project" to chanterelle, as well as lets you compile stuff with solc!

### CODE REVIEW

**Comment by iostat** [comment]
image
press F to pay respects

File: src/Compile.purs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+newtype SolcOutput =
+  SolcOutput { errors :: Maybe (Array SolcError)
+             , contracts :: M.StrMap OutputContract -- mapping of Filepath
+             }
+
+{-
+
+
+{ "contractName": "SimpleStorage"
```
Comment (blinky3713) [inline src/Compile.purs]: can we remove?

Comment (blinky3713) [reply src/Compile.purs]: I meant the comment

Comment (iostat) [reply src/Compile.purs]: oh that’s the big comment blob. showed up as the HACKHACKHACK line for me :/. keep it in, will help with making the types truffle compatible for the aforementioned

**Pull request merged**